tests/foreman/ui/test_organization.py

```diff
@@ -289,3 +329,74 @@ def test_positive_download_debug_cert_after_refresh(session):
                 session.subscription.refresh_manifest()
     finally:
         entities.Subscription(organization=org).delete_manifest(data={'organization_id': org.id})
+
+
+@pytest.mark.tier2
+def test_negative_errata_view_organization_switch(
+    session, module_org, module_lce, module_repos_col, request, default_sat
+):
+    """Verify no errata list visible on Organization switch
+
+    :id: a5317b2d-2593-4bee-9b20-78066cf2964c
+
+    :Steps: Create an Organization having a product synced which contains errata.
+
+    :expectedresults: Verify that the errata belonging to one Organization is not
+                      showing in the Default organization.
+
+    :CaseImportance: Critical
+
+    :CaseLevel: Integration
+    """
+    rc = RepositoryCollection(repositories=[YumRepository(settings.repos.yum_3.url)])
+    rc.setup_content(module_org.id, module_lce.id)
+    with session:
+        assert (
+            session.errata.search(CUSTOM_REPO_ERRATA_ID, applicable=False)[0]['Errata ID']
+            == CUSTOM_REPO_ERRATA_ID
+        )
+        session.organization.select(org_name="Default Organization")
+        assert not session.errata.search(CUSTOM_REPO_ERRATA_ID, applicable=False)
+
+
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.robottelo.REPOS_HOSTING_URL), reason='Missing repos_hosting_url')
+def test_negative_product_view_organization_switch(session, module_org):
+    """Verify product created in one organization is not visible in another
+
+    :id: 47f3580e-6071-4499-bf27-1ded62a7d3e3
+
+    :Steps:
+            1. Create an Organization having a product and verify that product is present in
+               the Organization.
+            2. Switch the Organization to default and verify that product is not visible in it.
+
+
+    :expectedresults: Verify that the Product belonging to one Organization is not visible in
+                      another organization.
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    product_name = gen_string('alpha')
+    product_label = gen_string('alpha')
+    product_description = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=read_data_file(VALID_GPG_KEY_FILE), organization=module_org
+    ).create()
```

Comment:
This all seems like something that should be done during test setup phase, in a fixture, as the test case is trying to provide coverage for the UI behavior when switching organization selection.

This also is a test that should fall under RBAC, are you sure we don't have coverage for it there?

Reply:
Updated.